Mage.Sets/src/mage/cards/t/TrespassersCurse.java

```diff
@@ -100,7 +100,8 @@ public boolean checkTrigger(GameEvent event, Game game) {
         Permanent enchantment = game.getPermanent(this.getSourceId());
         if (enchantment != null
                 && enchantment.getAttachedTo() != null
-                && game.getControllerId(event.getSourceId()).equals(enchantment.getAttachedTo())) {
+                && game.getControllerId(event.getSourceId()).equals(enchantment.getAttachedTo())
+                && game.getPermanent(this.getSourceId()).isCreature()) {
```

Comment:
I think that `this`, in this context, might in fact be the enchantment itself (which is never going to be a creature, thus the enchantment never triggers). Could be totally wrong (I've never programmed in Java!) - keep up the contributions however :)